source/app/ImmuneMLApp.py

```diff
@@ -75,13 +132,33 @@ def run_immuneML(namespace: argparse.Namespace):
 
 
 def main():
+    """Execute immuneML."""
     parser = argparse.ArgumentParser(description="immuneML command line tool")
     parser.add_argument("specification_path", help="Path to specification YAML file. Always used to define the analysis.")
     parser.add_argument("result_path", help="Output directory path.")
     parser.add_argument("--tool", help="Name of the tool which calls immuneML. This name will be used to invoke appropriate API call, "
                                        "which will then do additional work in tool-dependent way before running standard immuneML.")
-    run_immuneML(parser.parse_args())
-
+    parser.add_argument('-V', '--version', action='version',
+                        version='{} {}'.format(PROGRAM_NAME, VERSION))
+
+    args_dict = vars(parser.parse_args())
+
+    input_file = args_dict['specification_path']
+    output_file = args_dict['result_path']
+    # Store directories:
+    cwd_dir = os.getcwd()
+    input_dir = os.path.dirname(input_file)
+    if not os.path.isdir(input_dir):
+        input_dir = os.getcwd()
+
+    hello_world(input_file, cwd_dir, output_file)
+
+    try:
+        run_immuneML(args_dict)
+    except Exception as error:
+        print('ERROR - execution stopped.')
```

Comment:
This is a bad idea, if immuneML now encounters any error, the real error message is never printed
 and replaced by a standard printed message. Furthermore, it will look like execution did not fail (exit code 0), so in for example Galaxy the history element will show up green as if it executed correctly. I think the try/except statement should be removed.

Reply:
I dunno how galaxy work, but I guess the various constrains can be met.
This syntax allows to have a consistent opening and closing message. The error message needs to be shown and detailed in a error log file separated from the standard out (e.g. stderr and stdout, it is not now as you said, but first I wanted to be sure that this syntax doesn't clash with anything else) The logger links can be added right after the except, such it is consistent in all the code. 
One more reason/question,  isn't immuneML allowed to fail ? (E.g. some tests/example/unit test designed to fail?)

Reply:
I think it is not of absolute necessity for the goodbye message to be printed if immuneML fails, the most important thing in that case is understanding the error and easily being able to read it. A user might go through many rounds of error while trying to write up the specification file, then it should just be right there clearly visible and not be obstructed by other messages

Reply:
ps. thank you for the feedback, and please consider that this is more a way for me to see how immuneML works and you guys too, so I started from something not really important and relatively safe (well, that was the hope, the inconsistencies with galaxy, pip and pipy, logger and versions were not expected).  All the code added are just suggestions, I think they might be useful once smoothed out, but I am totally fine if you think that they won't help.

Reply:
totally! Let me a bit more time then to put together the log report as I was thinking to get the full info out. Like this is not helping, I agree.

Reply:
ps. just please let me know if that will conflict with something else in the code. Is it an useful update or?